packages/shorebird_cli/test/src/commands/releases/list_releases_command_test.dart

```diff
@@ -0,0 +1,160 @@
+
+    setUp(() {
+      auth = _MockAuth();
+      codePushClient = _MockCodePushClient();
+      httpClient = _MockHttpClient();
+      logger = _MockLogger();
+      command = ListReleasesCommand(
+        auth: auth,
+        logger: logger,
+        buildCodePushClient: ({
+          required httpClient,
+          hostedUri,
+        }) =>
+            codePushClient,
+      );
+
+      when(() => auth.client).thenReturn(httpClient);
+      when(() => auth.isAuthenticated).thenReturn(true);
+    });
+
+    test('description is correct', () {
+      expect(command.description, equals('List all releases for this app.'));
+    });
+
+    test('returns ExitCode.noUser when not logged in', () async {
+      when(() => auth.isAuthenticated).thenReturn(false);
+      expect(await command.run(), ExitCode.noUser.code);
+    });
+
+    test('returns ExitCode.config when shorebird is not initialized', () async {
+      final exitCode = await command.run();
+
+      verify(
+        () => logger.err(
+          any(
+            that: stringContainsInOrder([
+              'Shorebird is not initialized. Did you run',
+              'shorebird init',
+            ]),
+          ),
+        ),
+      ).called(1);
+      expect(exitCode, ExitCode.config.code);
+    });
+
+    test('returns ExitCode.software when unable to get releases', () async {
+      when(() => codePushClient.getReleases(appId: any(named: 'appId')))
+          .thenThrow(Exception());
+      final tempDir = setUpTempDir();
+
+      final exitCode = await IOOverrides.runZoned(
+        command.run,
+        getCurrentDirectory: () => tempDir,
+      );
+
+      expect(exitCode, ExitCode.software.code);
+    });
+
+    test('returns ExitCode.success when releases is empty', () async {
+      when(() => codePushClient.getReleases(appId: appId))
+          .thenAnswer((_) async => []);
+      final tempDir = setUpTempDir();
+
+      final exitCode = await IOOverrides.runZoned(
+        command.run,
+        getCurrentDirectory: () => tempDir,
+      );
+
+      expect(exitCode, ExitCode.success.code);
+      verify(() => logger.info('(empty)')).called(1);
+    });
+
+    test('returns ExitCode.success and prints releases when releases exist',
+        () async {
+      final tempDir = setUpTempDir();
+
+      when(() => codePushClient.getReleases(appId: appId)).thenAnswer(
+        (_) async => [
+          const Release(
+            id: 1,
+            appId: appId,
+            version: '1.0.1',
+            displayName: 'First',
+          ),
+          const Release(
+            id: 1,
+            appId: appId,
+            version: '1.0.2',
+            displayName: null,
+          ),
+        ],
+      );
+
+      final exitCode = await IOOverrides.runZoned(
+        command.run,
+        getCurrentDirectory: () => tempDir,
+      );
+
+      expect(exitCode, ExitCode.success.code);
+      verify(
+        () => logger.info('''
+┌───────────┬─────────┐
+│ Name      │ Version │
```

Comment:
Minor but might want to switch the order to show version first since that is guaranteed to be set

Reply:
Done.